### DISCUSSION TITLE
Issues in running sctp multi homing tests

### DISCUSSION THREAD

**Discussion by megarajan** [loxilb-io/loxilb#895] [General]
hi 

I am trying to verify sctp multihoming tests and running the first test case validation1.sh 
the testbed config is done as below

msivapra@dra-ldap:~/loxilb/sctp-mh$ docker ps
CONTAINER ID   IMAGE                              COMMAND       CREATED        STATUS        PORTS                            NAMES
08c78f1c63ec   ghcr.io/loxilb-io/nettest:latest   "/bin/bash"   17 hours ago   Up 17 hours                                    user
13e0c10a8125   ghcr.io/loxilb-io/nettest:latest   "/bin/bash"   17 hours ago   Up 17 hours                                    sw2
a3c9ec9e5971   ghcr.io/loxilb-io/nettest:latest   "/bin/bash"   17 hours ago   Up 17 hours                                    sw1
2e42b3ab334c   ghcr.io/loxilb-io/nettest:latest   "/bin/bash"   17 hours ago   Up 17 hours                                    r4
32595d1081d5   ghcr.io/loxilb-io/nettest:latest   "/bin/bash"   17 hours ago   Up 17 hours                                    r3
f0cec2755ba3   ghcr.io/loxilb-io/nettest:latest   "/bin/bash"   17 hours ago   Up 17 hours                                    r2
bbd988bdb77c   ghcr.io/loxilb-io/nettest:latest   "/bin/bash"   17 hours ago   Up 17 hours                                    r1
16792156d962   ghcr.io/loxilb-io/nettest:latest   "/bin/bash"   17 hours ago   Up 17 hours                                    ep1
64210e4cd3e1   ghcr.io/loxilb-io/loxilb:latest    "/bin/bash"   17 hours ago   Up 17 hours   3784/tcp, 11111/tcp, 22222/tcp   llb2
09fdb1bf7562   ghcr.io/loxilb-io/loxilb:latest    "/bin/bash"   17 hours ago   Up 17 hours   3784/tcp, 11111/tcp, 22222/tcp   llb1

I ran the validation1.sh and its failing in below line. 

msivapra@dra-ldap:~/loxilb/sctp-mh$  sudo ip netns exec llb1 curl -sX GET http://0.0.0.0:11111/netlox/v1/config/cistate/all -H 'accept: application/json'
[sudo] password for msivapra:

the above request returns null output so the tests are failing.

any clues on what I might be missing ?

**Comment by megarajan** [comment]
root@09fdb1bf7562:/# loxicmd get lb -o wide
|    EXT IP     |           SEC IPS            | HOST | PORT | PROTO |  NAME   | MARK | SEL |  MODE   |  ENDPOINT  | EPORT | WEIGHT | STATE | COUNTERS |
|---------------|------------------------------|------|------|-------|---------|------|-----|---------|------------|-------|--------|-------|----------|
| 10.10.10.10   |                              |      |   80 | tcp   |         |    0 | rr  | default | 31.31.31.1 |  8080 |      1 | -     | 0:0      |
| 11.11.11.11   |                              |      |   80 | tcp   |         |    0 | rr  | default | 31.31.31.1 |  8080 |      1 | -     | 0:0      |
| 123.123.123.1 | 124.124.124.1, 125.125.125.1 |      | 2020 | sctp  | sctpmh1 |    0 | rr  | fullnat | 31.31.31.1 |  9999 |      1 | -     | 0:0      |
| 133.133.133.1 | 134.134.134.1, 135.135.135.1 |      | 2020 | sctp  | sctpmh2 |    0 | rr  | fullnat | 1.1.1.1    |  9999 |      1 | -     | 0:0      |
root@09fdb1bf7562:/# l

root@09fdb1bf7562:/# loxicmd get ct help
| SERVICE NAME |   DESTIP   |   SRCIP    | DPORT | SPORT | PROTO |  STATE  | ACT | PACKETS |  BYTES   |
|--------------|------------|------------|-------|-------|-------|---------|-----|---------|----------|
| -            | 1.1.1.1    | 11.11.11.1 |  9999 | 12346 | sctp  | abort   |     |       0 |        0 |
| -            | 1.1.1.1    | 11.11.11.1 | 38547 | 38547 | icmp  | closed  |     |       0 |        0 |
| -            | 10.10.10.1 | 31.31.31.1 | 12346 |  9999 | sctp  | abort   |     |   15474 |   557064 |
| -            | 10.10.10.1 | 31.31.31.1 | 18127 | 18127 | icmp  | closed  |     |       2 |      256 |
| -            | 11.11.11.1 | 1.1.1.1    | 12346 |  9999 | sctp  | abort   |     |   15474 |   557064 |
| -            | 11.11.11.1 | 1.1.1.1    | 38547 | 38547 | icmp  | closed  |     |       2 |      256 |
| -            | 172.17.0.2 | 172.17.0.3 |  3784 | 56310 | udp   | udp-uni |     |  619236 | 32200272 |
| -            | 172.17.0.3 | 172.17.0.2 | 56310 |  3784 | udp   | udp-uni |     |       0 |        0 |
| -            | 31.31.31.1 | 10.10.10.1 |  9999 | 12346 | sctp  | abort   |     |       0 |        0 |
| -            | 31.31.31.1 | 10.10.10.1 | 18127 | 18127 | icmp  | closed  |     |       0 |        0 |
root@09fdb1bf7562:/#

**Comment by megarajan** [comment]
root@09fdb1bf7562:/# loxicmd get  ip
| DEVICE NAME |  IP ADDRESS   |
|-------------|---------------|
| eth0        | 172.17.0.2/16 |
| lo          | 127.0.0.1/8   |
| vlan10      | 10.10.10.1/24 |
| vlan11      | 11.11.11.1/24 |
root@09fdb1bf7562:/# loxicmd get session
root@09fdb1bf7562:/# loxicmd get endpoint
|    HOST    |         NAME         | PTYPE | PORT | DURATION | RETRIES | MINDELAY  | AVGDELAY  | MAXDELAY  | STATE |
|------------|----------------------|-------|------|----------|---------|-----------|-----------|-----------|-------|
| 1.1.1.1    | 1.1.1.1_sctp_8080    | ping: |    0 |       60 |       2 | 237.523µs | 262.291µs | 287.058µs | ok    |
| 1.1.1.1    | 1.1.1.1_sctp_9999    | sctp: | 9999 |       60 |       2 |           |           |           | nok   |
| 31.31.31.1 | 31.31.31.1_sctp_8080 | ping: |    0 |       60 |       2 | 193.905µs | 326.078µs | 458.25µs  | ok    |
| 31.31.31.1 | 31.31.31.1_tcp_8080  | none: | 8080 |        0 |       0 |           |           |           | ok    |
| 31.31.31.1 | 31.31.31.1_sctp_9999 | sctp: | 9999 |       60 |       2 |           |           |           | nok   |
root@09fdb1bf7562:/# loxicmd get hastate
| INSTANCE | HASTATE |
|----------|---------|
| default  | BACKUP  |
root@09fdb1bf7562:/#

**Comment by TrekkieCoder** [comment]
We have gh-actions cicd for sctp-multihoming with kubernetes [here](https://github.com/loxilb-io/loxilb/actions/workflows/k3s-sctpmh.yml) . Please check the respective yaml file and scripts [here](https://github.com/loxilb-io/loxilb/actions/workflows/k3s-sctpmh.yml).
